src/http.rs

```diff
@@ -31,10 +32,15 @@ pub fn get_client() -> Client<Connector, hyper::Body> {
 pub fn fetch_sync_string(module_name: &str) -> DenoResult<String> {
   let url = module_name.parse::<Uri>().unwrap();
   let client = get_client();
-  let future = client
-    .get(url)
-    .and_then(|response| response.into_body().concat2());
-  let body = tokio_util::block_on(future)?;
+  let get_future = client.get(url);
+  let response = tokio_util::block_on(get_future)?;
+  if !response.status().is_success() {
+    Err(io::Error::new(
+      io::ErrorKind::NotFound,
+      format!("cannot load from '{}'", module_name),
+    ))?;
+  }
+  let body = tokio_util::block_on(response.into_body().concat2())?;
```

Comment:
Maybe it doesn't matter but this could be done using only one `block_on`. It would be nice to have a test showing we get NotFound error when trying to hit 404 url.

Reply:
So I have somehow combined into a single `block_on`, but had to explicitly implement Future for a struct holding `Concat2` and `StatusCode` (to implement using enum that either holds body or and error seems more problematic)
(I started to use Rust only very recently, so this might not be a very good solution from the viewpoint of elegance. Would like to learn about better alternatives!)

btw a NotFound test is added